ibis/expr/operations.py

```diff
@@ -2403,9 +2407,9 @@ class ArraySlice(ValueOp):
     input_type = [
         rules.array(dt.any),
         rules.integer(name='start'),
-        rules.integer(name='stop')
+        rules.integer(name='stop', optional=True)
```

Comment:
`step` isn't supported yet. We could make the implementation specific to the backend and have the translation rules raise for SQL backends and have pandas work. I'll make an issue.